### DISCUSSION TITLE
Localisation / Translation Updates

### DISCUSSION THREAD

**Discussion by lrljoe** [rappasoft/laravel-livewire-tables#1956] [General]
To avoid conflicts with other packages, and core localisations, we will shortly be prefixing localisations with "livewire-tables", in the same way that "views" etc are prefixed.

This package has significant translations & localisations, which we as the core team are extremely grateful to those that contribute localisation/translations!
We actively encourage contributions for localisations, so please reach out to add your respective locale to the core package.

I am presently updating the localisation files so that the package prefix is present, e.g.
**Present**
```js
{
    "All": "Tous",
    "All Columns": "Toutes les colonnes",
    "Applied Filters": "Filtres appliqués",
  // Etc
}
```

**New**
This then changes to be prefixed with the package, to mitigate any conflicts, for example:
```js
{
    "livewire-tables::All": "All",
    "livewire-tables::All Columns": "All Columns",
    "livewire-tables::Applied Filters": "Applied Filters",
  // Etc
}
```

Or In the fr.json file:
```js
{
    "livewire-tables::All": "Tous",
    "livewire-tables::All Columns": "Toutes les colonnes",
    "livewire-tables::Applied Filters": "Filtres appliqués",
// etc
}
```

I'm updating the mappings in the files for the next release.  

In addition, we will be adding in some tests to ensure that the localised versions of strings are present, while these won't prevent a release, I'd ideally like to have a channel that helps everyone keep up-to-date well ahead of any official release.

Therefore - I'd be extremely grateful if you could please give me a nudge on the official Discord so that I know who you are!  This will help me to reach out when we have any significant changes/language additions.

@irineujunior
@jeppeolesen
@siebsie23
@spekkie2002
@Jerimu
@devmikromike
@dgillier
@wanadri
@meavric
@CristhoferMF
@Oleksandr-Moik

**Comment by meavric** [comment]
Hi,  I joined to &#34;team&#34; in Discord  BR   
     
      
       
        Dnia 20 września 2024 02:00 Joe ***@***.***&gt; napisał(a):
       
    
       
         
 To avoid conflicts with other packages, and core localisations, we will shortly be prefixing localisations with &#34;livewire-tables&#34;, in the same way that &#34;views&#34; etc are prefixed. 
 This package has significant translations &amp; localisations, which we as the core team are extremely grateful to those that contribute localisation/translations!  
We actively encourage contributions for localisations, so please reach out to add your respective locale to the core package. 
 I am presently updating the localisation files so that the package prefix is present, e.g.  
 Present 
 { 
     &#34;All&#34; :  &#34;Tous&#34; , 
     &#34;All Columns&#34; :  &#34;Toutes les colonnes&#34; , 
     &#34;Applied Filters&#34; :  &#34;Filtres appliqués&#34; , 
   // Etc 
 } 
 New  
This then changes to be prefixed with the package, to mitigate any conflicts, for example: 
 { 
     &#34;livewire-tables::All&#34; :  &#34;All&#34; , 
     &#34;livewire-tables::All Columns&#34; :  &#34;All Columns&#34; , 
     &#34;livewire-tables::Applied Filters&#34; :  &#34;Applied Filters&#34; , 
   // Etc 
 } 
 Or In the fr.json file: 
 { 
     &#34;livewire-tables::All&#34; :  &#34;Tous&#34; , 
     &#34;livewire-tables::All Columns&#34; :  &#34;Toutes les colonnes&#34; , 
     &#34;livewire-tables::Applied Filters&#34; :  &#34;Filtres appliqués&#34; , 
 // etc 
 } 
 I&#39;m updating the mappings in the files for the next release.  However- I&#39;d be extremely grateful if you could please give me a nudge on the official Discord so that I know who you are!  This will help me to reach out when we have any significant changes/language additions. 
 github.com @irineujunior  
 github.com @jeppeolesen  
 github.com @siebsie23  
 github.com @spekkie2002  
 github.com @Jerimu  
 github.com @devmikromike  
 github.com @dgillier  
 github.com @wanadri  
 github.com @meavric  
 github.com @CristhoferMF  
 github.com @Oleksandr-Moik

**Reply by lrljoe** [reply]
Please give me a nudge on Discord with your github username, so I can keep track of who you are!

**Comment by Aiiion** [comment]
I realize why you are choosing to go down this path but it should be configurable and not hard coded within the view files. As in you should be able to choose the localization string being used to fetch the global localization strings within the system.
 
Now instead of showing the English equivalent, if a translation is missing, you are showing "livewire-tables::Filters" for example. Thereby breaking the copy for users where there is no translation within this package.

I am currently having this issue, I will submit a PR with the translations, But if there is a work around advise that would be highly appreciated.

**Reply by Aiiion** [reply]
Ok I am having a bit of a day ofc there is a simple workaround. I will submit swedish translations in a bit

**Reply by lrljoe** [reply]
Fair enough, please see my comments on your PR, as the "development" branch is what needs to be targeted for a PR (and I don't want to break your branch by forcing an update)

We really appreciate any localisation contributions btw!
